static/js/util/profile_edit.js

```diff
@@ -49,36 +55,66 @@ export function boundSelectField(keySet, label, options) {
     errors,
     updateProfile,
   } = this.props;
-  let onChange = value => {
+  let onChange = option => {
     let clone = _.cloneDeep(profile);
-    _.set(clone, keySet, value? value.value : '');
+    let toStore;
+    if (option.value !== null && option.value !== undefined) {
+      toStore = option.value.props.value;
```

Comment:
I'd rather not since we can't call it `option.value` and I can't think of a better name for it. It also means I'd have to update the current unit tests so that `onChange` can make use of it